lib/scholar/neighbors/random_projection_forest.ex

```diff
@@ -130,17 +170,16 @@ defmodule Scholar.Neighbors.RandomProjectionForest do
   defn fit_n(tensor, key, opts) do
     depth = opts[:depth]
     num_trees = opts[:num_trees]
-    type = to_float_type(tensor)
     {size, dim} = Nx.shape(tensor)
     num_nodes = 2 ** depth - 1
 
     {hyperplanes, _key} =
-      Nx.Random.normal(key, type: type, shape: {num_trees, depth, dim})
+      Nx.Random.normal(key, type: :f64, shape: {num_trees, depth, dim})
```

Comment:
I don't think we should default to f64, What if we allow the type as an argument? So we do: `type = opts[:type] || to_float_type(tensor)`?

Reply:
It seems that initializing hyperplanes and medians to `f64` might not be necessary. I changed the code in `predict/2` from 
```
proj = Nx.dot(query, hyperplanes[level])
```
(hyperplanes were vectorized by first coordinates which corresponds to trees) to
```
proj = Nx.dot(hyperplanes[[.., level]], [1], query, [1]) |> Nx.vectorize(:trees)
```
which is kinda the same as done in `fit/2`. Now the projections are the same inside `fit/2` and `predict/2`, even with `f32`; it seems the order of the arguments was important as well. Should have done this before, sorry about that!
It will work for constructing the k-NN graph, i.e. when `query = tensor`, where `tensor` is used to grow the forest. Might not work for arbitrary query (i.e. there might be some issues with projections as described in https://github.com/elixir-nx/scholar/issues/230). Perhaps adding a note in the docs might be a good idea?